kof-operator/internal/k8s/cluster_deployment.go

```diff
@@ -0,0 +1,16 @@
+package k8s
+
+import (
+	"context"
+
+	kcmv1alpha1 "github.com/K0rdent/kcm/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetClusterDeployments(ctx context.Context, client client.Client) (*kcmv1alpha1.ClusterDeploymentList, error) {
+	cdList := &kcmv1alpha1.ClusterDeploymentList{}
+	if err := client.List(ctx, cdList); err != nil {
+		return nil, err
```

Comment:
In respect to this [comment](https://github.com/k0rdent/kof/pull/275/files#r2104498628), probably better to return a non-nil cdList with empty Items.
```suggestion
		return cdList, err
```